### PR TITLE
Support softmax LSE in causal and local masking

### PR DESCRIPTION
**Pull request by polisettyvarma**

### CODE REVIEW

File: python/sgl_kernel/flash_attn.py [hunk 1]
```diff
@@ -214,6 +214,9 @@ def flash_attn_with_kvcache(
            to automatically determine the number of splits.
            Don't change this unless you know what you are doing.
         return_softmax_lse: bool. Whether to return the logsumexp of the attention scores.
+            Supported alongside causal and local (sliding-window) masking. Not supported
+            with sink logits (``sinks``) or an fp8 KV cache (no LSE kernel instantiation
+            exists for those); requesting it there raises a ``RuntimeError``.
```
Comment (Valentine233) [inline python/sgl_kernel/flash_attn.py]: Relative attention is also not supported.

File: src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp [hunk 2]
```diff
@@ -402,7 +402,21 @@ class XeFMHAFwdKernel {
...
+          // unconditionally, so the softmax denominator stays 0 and the epilogue
+          // writes the all-zero output (and -inf softmax_lse) these rows must
+          // hold. Skipping the tile would leave the caller's `out` rows
+          // untouched, i.e. uninitialized -- with seq_len_qo > seq_len_kv_cache
+          // and a bounded window that is most of the output. The causal path
+          // relies on the same zero-iteration behaviour (seq_len <= 0 there).
+          // Use [0,0) rather than [blk_k1,blk_k1) so the paged K prefetch's
+          // page-table lookup at blk_k0 stays in bounds.
+          blk_k0 = 0;
+          blk_k1 = 0;
```
Comment (Valentine233) [inline src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp]:
```suggestion
          if constexpr (LSE) {
            blk_k0 = 0;
            blk_k1 = 0;
          }
```

Comment (Valentine233) [reply src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp]: Remain the perf for cases without LSE.

File: src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp [hunk 3]
```diff
@@ -402,7 +402,21 @@ class XeFMHAFwdKernel {
...
-        if (blk_k0 >= blk_k1) continue;
+        if (blk_k0 >= blk_k1) {
+          // No K block intersects this Q tile's sliding window, so every row in
+          // the tile attends to nothing. Run the mainloop over an EMPTY K range
+          // instead of skipping the tile: the mainloop clears the accumulators
+          // unconditionally, so the softmax denominator stays 0 and the epilogue
+          // writes the all-zero output (and -inf softmax_lse) these rows must
+          // hold. Skipping the tile would leave the caller's `out` rows
+          // untouched, i.e. uninitialized -- with seq_len_qo > seq_len_kv_cache
+          // and a bounded window that is most of the output. The causal path
```
Comment (Valentine233) [inline src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp]: As this is inside `LocalMask` loop, why to mention causal path here?

File: tests/test_flash_attention.py
Comment (Valentine233) [inline tests/test_flash_attention.py:514]: For the case of causal and local masking, the check of `-inf` is necessary. Please add:
```
expected_neginf = torch.isneginf(ref_lse_hq)
assert torch.isneginf(lse_f[expected_neginf]).all()
```

File: python/sgl_kernel/flash_attn.py (hunk 1 above)
Comment (polisettyvarma) [reply python/sgl_kernel/flash_attn.py]: done

File: src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp (hunk 2 above)
Comment (polisettyvarma) [reply src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp]: done

File: tests/test_flash_attention.py
Comment (polisettyvarma) [reply tests/test_flash_attention.py:514]: done

File: src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp (hunk 3 above)
Comment (polisettyvarma) [reply src/sycl/kernels/flash_attention_v2/kernel/xe_fmha_fwd_kernel.hpp]: done

**Review by Valentine233 — APPROVED** [review]
